### PR TITLE
Adding transaction for embedding

### PR DESCRIPTION
**Pull request by dipinknair**
Adding Transaction for embedding till https://github.com/ansys/ansys-pythonnet/issues/14 gets fixed.

usage
``` 
with Transaction():
    pass
 ```

### CODE REVIEW

File: src/ansys/mechanical/core/embedding/transaction.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        """Enter the context of the transaction."""
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Exit the context of the transaction and disposes of resources."""
+        if not self._disposed:
+            self._dispose()
+
+    def _dispose(self):
+        if self._disposed:
```
Comment (koubaa) [inline src/ansys/mechanical/core/embedding/transaction.py:65]: the self._disposed pattern is not necessary. The C# class itself will do these checks. You can call Dispose() as many times as you want on an IDisposable!

File: src/ansys/mechanical/core/embedding/imports.py [hunk 2]
```diff
@@ -51,8 +52,9 @@ def global_variables(app: "ansys.mechanical.core.App", enums: bool = False) -> t
     vars["Quantity"] = Quantity
     vars["System"] = System
     vars["Ansys"] = Ansys
-    vars["Transaction"] = Transaction
-
+    # vars["Transaction"] = Transaction
+    # When ansys-pythonnet issue #14 is fixed, uncomment above
+    # and Transaction class will be removed
```
Comment (koubaa) [inline src/ansys/mechanical/core/embedding/imports.py:57]: this is not quite right, Transaction needs to go in the globals!

File: src/ansys/mechanical/core/embedding/transaction.py (hunk 1 above)
Comment (dipinknair) [reply src/ansys/mechanical/core/embedding/transaction.py]: I thought you suggested to remove the _dispose function.  I kept it for the test. if not necessary please let me know

File: src/ansys/mechanical/core/embedding/imports.py [hunk 3]
```diff
@@ -71,3 +72,34 @@ def get_all_enums() -> typing.Dict[str, typing.Any]:
...
+
+    def __init__(self):
+        """Initialize the Transaction class."""
+        import clr
+
+        clr.AddReference("Ansys.ACT.WB1")
+        import Ansys
+
+        self._transaction = Ansys.ACT.Mechanical.Transaction()
+        self._disposed = False
```
Comment (koubaa) [inline src/ansys/mechanical/core/embedding/imports.py:96]: _disposed is not necessary here

File: tests/embedding/test_globals.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    assert "Ansys" in globals_dict
+    assert "Transaction" in globals_dict
+
+
+@pytest.mark.embedding
+def test_transaction(embedded_app):
+    """Test Transaction"""
+    globals().update(pymechanical.global_variables(embedded_app, True))
+
+    with Transaction() as transaction:
```
Comment (koubaa) [inline tests/embedding/test_globals.py:49]: I don't think this test is necessary

File: tests/embedding/test_globals.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    """Test the global variables"""
+    globals_dict = global_variables(embedded_app, True)
+    assert "ExtAPI" in globals_dict
+    assert "DataModel" in globals_dict
+    assert "Model" in globals_dict
+    assert "Tree" in globals_dict
+    assert "Quantity" in globals_dict
+    assert "System" in globals_dict
+    assert "Ansys" in globals_dict
+    assert "Transaction" in globals_dict
```
[the remark this answers is not in the document]
Comment (dipinknair) [reply tests/embedding/test_globals.py:41]: Yes

**Review by koubaa — APPROVED** [review]

**Pull request merged**